esrally/mechanic/telemetry.py

```diff
@@ -431,14 +431,17 @@ def record_stats_per_index(self, name, stats):
 
         for shard_stats in stats:
             if "shard_id" in shard_stats:
+                doc = {
+                    "name": "ccr-stats",
+                    "shard": shard_stats
+                }
                 shard_metadata = {
                     "cluster": self.cluster_name,
                     "index": name,
```

Comment:
This seems to be included in the shard stats already. Does it make sense to duplicate this information?

Reply:
The shard metrics, as reported by the CCR stats endpoint are associated with each following index and I wanted to retain the structure in this respect.

For example in the following excerpt, we can see this duplication, a key `index` and then per shard the more specific `leader-index` / `follower_index` keys. I feel that for the sake of maintaining this structure it makes sense to keep this duplication.

```
   "indices" : [
      {
        "index" : "geonames-copy",
        "shards" : [
          {
            "remote_cluster" : "leader",
            "leader_index" : "geonames",
            "follower_index" : "geonames-copy",
            "shard_id" : 0,
            "leader_global_checkpoint" : -1,
            "leader_max_seq_no" : -1,
            "follower_global_checkpoint" : -1,
            "follower_max_seq_no" : -1,
            "last_requested_seq_no" : -1,
            "outstanding_read_requests" : 1,
            "outstanding_write_requests" : 0,
            "write_buffer_operation_count" : 0,
            "write_buffer_size_in_bytes" : 0,
            "follower_mapping_version" : 1,
            "follower_settings_version" : 1,
            "total_read_time_millis" : 0,
            "total_read_remote_exec_time_millis" : 0,
            "successful_read_requests" : 0,
            "failed_read_requests" : 0,
            "operations_read" : 0,
            "bytes_read" : 0,
            "total_write_time_millis" : 0,
            "successful_write_requests" : 0,
            "failed_write_requests" : 0,
            "operations_written" : 0,
            "read_exceptions" : [ ],
            "time_since_last_read_millis" : 4004
          },
          {
            "remote_cluster" : "leader",
            "leader_index" : "geonames",
            "follower_index" : "geonames-copy",
            "shard_id" : 1,
            "leader_global_checkpoint" : -1,
            "leader_max_seq_no" : -1,
            "follower_global_checkpoint" : -1,
            "follower_max_seq_no" : -1,
            "last_requested_seq_no" : -1,
            "outstanding_read_requests" : 1,
            "outstanding_write_requests" : 0,
            "write_buffer_operation_count" : 0,
            "write_buffer_size_in_bytes" : 0,
            "follower_mapping_version" : 1,
            "follower_settings_version" : 1,
            "total_read_time_millis" : 0,
            "total_read_remote_exec_time_millis" : 0,
            "successful_read_requests" : 0,
            "failed_read_requests" : 0,
            "operations_read" : 0,
            "bytes_read" : 0,
            "total_write_time_millis" : 0,
            "successful_write_requests" : 0,
            "failed_write_requests" : 0,
            "operations_written" : 0,
            "read_exceptions" : [ ],
            "time_since_last_read_millis" : 4087
          },
```

Reply:
Sure, I'm fine with that.